src/test/java/org/mastodon/mamut/io/importer/labelimage/LabelImageUtilsTest.java

```diff
@@ -151,13 +149,119 @@ void testCreateSpotFromWrongVoxelDimensions()
 	}
 
 	@Test
-	void testImportSpotsFromBdvChannel()
+	void testImportSpotSphere()
 	{
-		LegacyInjector.preinit();
 		try (Context context = new Context())
 		{
-			int pixelValue = 1;
-			Img< FloatType > img = createImageCubeCorners( pixelValue );
+			Img< FloatType > img = ArrayImgs.floats( 12, 12, 12 );
+			SphereRenderer.renderSphere( new int[] { 5, 5, 5 }, 5, 1, img );
+			ProjectModel projectModel = DemoUtils.wrapAsAppModel( img, model, context );
+			LabelImageUtils.importSpotsFromBdvChannel( projectModel, projectModel.getSharedBdvData().getSources().get( 0 ).getSpimSource(),
+					1, false );
+
+			Iterator< Spot > iter = model.getGraph().vertices().iterator();
+			Spot spot = iter.next();
+			double[][] covarianceMatrix = new double[ 3 ][ 3 ];
+			spot.getCovariance( covarianceMatrix );
+			final JamaEigenvalueDecomposition eigenvalueDecomposition = new JamaEigenvalueDecomposition( 3 );
+			eigenvalueDecomposition.decomposeSymmetric( covarianceMatrix );
+			final double[] eigenValues = eigenvalueDecomposition.getRealEigenvalues();
+			double semiAxisA = Math.sqrt( eigenValues[ 0 ] );
+			double semiAxisB = Math.sqrt( eigenValues[ 1 ] );
+			double semiAxisC = Math.sqrt( eigenValues[ 2 ] );
```

Comment:
Code duplication, this eigenvalue computation is copy pasted 3 times.

Reply:
I introduced a helper method to reduce to code duplicaton.